### DISCUSSION TITLE
Performance degradation since update

### DISCUSSION THREAD

**Discussion by ghost** [lllyasviel/stable-diffusion-webui-forge#2392] [Q&A]
Hello! There was an update on 27 November. Since then the Flux model generation with RTX 4060 has been very slow. It used to take 2-3 minutes per image and now it takes much longer. Windows freezes several times for minutes during generation. I have not experienced this before. Also loading XL models seems to be slowed down and I experience momentary freezing. Performance has degradation. I have reinstalled it and no improvement.

**Comment by aggregate15** [comment]
I also have 4060. For me, since the last update, the Generation times are exactly the same as before, but yes, moving the models is taking twice as much time now. Something is definitely wrong on model loading. 

However regarding generation times, can you share your settings for Image Generation ? I use full 8187 VRAM, Queue Swap Method and Shared Swap Location and generate images at 896/1152, it gives me the fastest generation times. At least in my case there is no difference with these settings. Maybe you can try the same settings and see how it goes ?

**Comment by ghost** [comment]
Model loading is slow and so is generation. At the resolution you posted, the first image takes 2 minutes. The second picture before it took a long time and 3:22 minutes. The third image has more prompt and 6:44 minutes. Fourth picture with 1 LoRa at 10:42 minutes and 95% stop and stuttering Windows. I did the test with the Flux1DevHyperNF4 model which is nearly 12GB. I have used the 16GB Flux Dev model without any problems, now it freezes the system. In this case I don't wait and restart the PC. I don't even dare to use LoRA. So far I have added 5 LoRA models to the Flux Dev 16GB.

**Reply by aggregate15** [reply]
No, something is definitely wrong in your case. For me a normal generation with these settings always took around 1:30 secs and with LORAs it goes up to about 1:50 mins at max and it pretty much stays the same even if I do 50 generations. I had lots of interactions with 4060 users in the past in many forums and they all get the same results. What is your system RAM ? I was throttling earlier at 16GB but it goes real smooth with 32GB RAM. Btw I use flux1-dev-fp8. It gives me faster results than NF4, etc. Also make sure you select "Automatic (fp16 LoRA)" . In the Diffusion in Low Bits, whenever you choose the LORAs. Lastly, go for the Forge with CUDA 12.4 + Pytorch 2.4 Download. It always works much faster. Let me know if any of this helps as this is all I could think of with a 4060.

**Reply by ghost** [reply]
I have 16GB ram and since the update, Forge has been running chaotically. Slow model loading, slow generation and windows system freezes. I also reinstalled Forge. For me the Flux Dev 16GB version produced much better quality and detail. The smaller versions also stutter. Ok, I will try it.

Update: You need a clip model, the ones I use don't work on my PC. the ones I use include the clip and t5 models or whatever you need. Which one do you recommend exactly?

**Reply by aggregate15** [reply]
There is no need for separate clip and VAE, etc, with the flux1-dev-fp8. Its 16GB and you can use it as a stand alone model. an I am sure the RAM must be your problem. After your post, I checked and noticed ... that the earlier Flux would taker around 22 GB of RAM, but with the new update, its taking upto to 28GB when using Flux. Your RAM must be throttling earlier, trying to fit in the model and now it would have become worse. Try upgrading your RAM if you can. It makes a huge difference. You'll be getting consistent FLUX results in 1:20 to 1:40, no problem

**Reply by ghost** [reply]
I used 16GB Flux and it was fine two days ago. Now the 12GB models are also having problems.

Update: The XL also stutters and suffers in some cases, two days ago there was no problem. If it happens to work again, I'll make a backup of it.

**Comment by ghost** [comment]
I put in my girlfriend's +16GB memory and I'm still not satisfied. The 16GB Flux is still not working, two days ago it was working with 16GB memory. Now with 16GB memory I am trying the bnb-nf4 setup and it seems fine. But I added a LoRA and it froze at 95%. Two days ago I used the 16GB Flux with 5 LoRA and it took 3 minutes to generate each time.

**Reply by ghost** [reply]
I am using this now and have no problems: flux1DevHyperNF4, DPM++ 2M, Karras, Diffusion in low bits: Automatic and 768x1024 resolution. Generation time is 1 minute. I can no longer use LoRA. I also added 5 LoRA to flux_dev 16Gb before the upgrade. 

Update: Now the 16GB flux_dev works without any problems. I haven't done anything, just been generating for many hours. I watch movies and surf the internet next to it, no freezing.
